### DISCUSSION TITLE
How to ensure there is only one array of spans per scope name per transaction?

### DISCUSSION THREAD

**Discussion by ramgdev** [open-telemetry/opentelemetry-rust#1862] [Q&A] [answered]
I am noticing the serialized json file shows multiple `scopeSpans` with the same scope name. Each of them include only one span in the array. I would like to avoid this and include all those spans into a single array. Because, each `scopeSpans` is also resulting in the `resource` attributes to be repeated. As a result, I end up having as many `resource` as there are spans. It would be good to have one `resource` for all the spans that are using the same scope name per transaction. I'm using v0.22.0.

**Comment by lalitb** [comment]
Are you using batch processor ? This should batch all the spans received in (default) 5 seconds, and send to exporter, which should eventually be serialized together. If not, this could be a bug somewhere.

**Comment by lalitb** [comment]
@ramgdev Sorry, I understand your question now. You are correct; the exporter should group all the events (spans/logs) under the same scope and, further, under the same resource together before exporting. I will create an issue for this - if someone wants to pick, I can help with the changes.

**Reply by ramgdev** [reply]
@lalitb sure, please assign the issue to me and I'll try to fix it with your guidance.

**Reply by lalitb** [reply, marked as answer]
Thanks, @ramgdev  - I have made some changes for the logs. Let me complete and create a PR for that, and you can pick similar changes for spans.

**Reply by ramgdev** [reply]
Marking this answered as @lalitb has addressed the changes in this PR https://github.com/open-telemetry/opentelemetry-rust/pull/1873
